javascript/cw_tool.path.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @file
+ *  Add path related behaviours.
+ */
+var CW = CW || {};
+
+(function($) {
+
+  'use strict';
+    
+  CW.Path = {
+
+    initialized: false,
+    
+    uri: document.location.href,
+    
+    urlProtocol: '',
+    urlHostname: '',
+    urlPath: '',
+    urlSearch: '',
+    urlQueryParam: [],
+    urlFragment: '',
+    
+    initializeIfNecessary: function() {
+      if( !this.initialized ) {
+        this.parseURI();
+        this.initialized = true;
+      }
+    },    
+    
+    getURI: function() {
+      return this.uri;
+    },
+    
+    setURI: function( uri ) {
+      this.uri = uri;
+    },
+    
+    getUrlProtocol: function() {
+      this.initializeIfNecessary();
+      return this.urlProtocol;  
+    },
+    
+    setUrlProtocol: function( val ) {
+      this.urlProtocol = val;
+    },
+    
+    getUrlHostname: function() {
+      this.initializeIfNecessary();
+      return this.urlHostname;  
+    },
+
+    getUrlBaseHostname: function() {
+      if (!this.isIPAddress(this.getUrlHostname())) {
+        var components = this.getUrlHostname().split('.');
+        return components.slice(-2).join('.');
+      }
+      return this.getUrlHostname();
+    },
+    
+    setUrlHostname: function( val ) {
+      this.urlHostname = val;
+    },
+    
+    getUrlPath: function() {
+      this.initializeIfNecessary();
+      return this.urlPath;  
+    },
+    
+    setUrlPath: function( val ) {
+      this.urlPath = val;
+    },
+    
+    getUrlSearch: function() {
+      this.initializeIfNecessary();
+      return this.urlSearch;  
+    },
+    
+    setUrlSearch: function( val ) {
+      this.urlSearch = val;
+    },
+    
+    getUrlFragment: function() {
+      this.initializeIfNecessary();
+      return this.urlFragment;  
+    },
+    
+    setUrlFragment: function( val ) {
+      this.urlFragment = val;
+    },
+    
+    //Parameters
+    parseURI: function() {
+      var urlElement = document.createElement('a');
+      urlElement.href = this.getURI();
```

Comment:
I don't see calling ```setURI``` anywhere which seems the only way the covered variable is set. How is this coordinated?